api/control/v1alpha1/types.go

```diff
@@ -394,27 +394,14 @@ const (
 	AggregatorTypeMax   AggregatorType = "MAX"
 )
 
-// Expression is some value to derive from a workload object from a WEC.
-// An Expression is either a JSONPath identifying a value to extract
-// or a boolean combination of comparisons of atomic values.
-// While the expression here in the Go type system admits other values,
-// the controller will accept only the restricted set stated above.
-// Post-apiserver validation errors are posted to the StatusCollectorStatus.
-type Expression struct {
-	Op   ExpressionOperator `json:"op"`
-	Path string             `json:"path,omitempty"`
-	Args []Expression       `json:"args,omitempty"`
-}
-
-type ExpressionOperator string
-
-const (
-	OperatorPath  ExpressionOperator = "Path" // JSONPath string
-	OperatorOr    ExpressionOperator = "Or"
-	OperatorAnd   ExpressionOperator = "And"
-	OperatorNot   ExpressionOperator = "Not"
-	OperatorEqual ExpressionOperator = "Equal"
-)
+// Expression is written in the Common Expression Language.
+// See github.com/google/cel-go/cel for the Go implementatiom used in Kubernetes.
+// The expression will be type-checked against the schema for the object type at hand,
+// using the Kubernetes library code for converting an OpenAPI schema to a CEL type
+// (e.g., https://github.com/kubernetes/apiserver/blob/v0.28.2/pkg/cel/common/schemas.go#L40).
+// Parsing errors are posted to the status.Errors of the StatusCollector.
+// Type checking errors are posted to the status.Errors of the Binding and BindingPolicy.
+type Expression string
```

Comment:
Can you update documentation to reflect the (base of the) expected schema?

Reply:
What do you mean by "base of the"?

Reply:
As in the first version of the schema, maybe a redundant phrase. The intention is to have the expected and supported schema documented for users to know. I assumed that it would be equivalent to how `Expression` was defined previously.

Reply:
I am still not sure I understand the question. The point of this PR is to change from the previous formulation of Expression to using CEL. See how the comment on this type says that the expression is written in the Common Expression Language?

Reply:
I confused it with something else. It may be useful to add a link to this - https://kubernetes.io/docs/reference/using-api/cel/  - either here or in the documentation.

Also, line 398: s/implementatiom/implementation